### ISSUE TITLE
Stage 4 completes before cephfs is mountable

### ISSUE THREAD

**Issue by smithfarm** [SUSE/DeepSea#321]
When running DeepSea from a script, I cannot mount the cephfs immediately after Stage 4 completes: I have to insert a delay. 

It would be nicer if Stage 4 could wait, and not complete until the cephfs is mountable.

Here is the relevant part of the test log:

```
+ echo '********** Stage 4 completed successefully **********'
+ cephfs_mount_and_sanity_test
********** Stage 4 completed successefully **********
++ ceph-conf --lookup mon_initial_members
++ tr -d '[:space:]'
+ local mons=target213032072133,target213032072130
++ grep 'key =' /etc/ceph/ceph.client.admin.keyring
++ awk '{print $NF}'
+ local secr=AQAqyTtZAAAAABAAzg5knVWdI74yPCGMIK0CZQ==
++ whoami
Running as: root
Mounting cephfs
MONs: target213032072133,target213032072130
admin secret: AQAqyTtZAAAAABAAzg5knVWdI74yPCGMIK0CZQ==
+ echo 'Running as: root'
+ echo 'Mounting cephfs'
+ echo 'MONs: target213032072133,target213032072130'
+ echo 'admin secret: AQAqyTtZAAAAABAAzg5knVWdI74yPCGMIK0CZQ=='
+ test -d /mnt
+ mount -t ceph target213032072133,target213032072130:/ /mnt -o name=admin,secret=AQAqyTtZAAAAABAAzg5knVWdI74yPCGMIK0CZQ==
mount error 2 = No such file or directory
```

When I insert a 10-second delay, it succeeds:

```
+ echo '********** Stage 4 completed successefully **********'
+ sleep 10
********** Stage 4 completed successefully **********
+ cephfs_mount_and_sanity_test

... (snip) ... 

+ test -d /mnt
+ mount -t ceph target213032073236,target213032073221:/ /mnt -o name=admin,secret=AQAazTtZAAAAABAAda+kE5hrHgETA71lcewyUg==
+ touch /mnt/bubba
+ test -f /mnt/bubba
+ umount /mnt 
OK
+ echo OK
```

**Comment by smithfarm** [comment]
@jan--f Any idea what's going on here?

**Comment by jan--f** [comment]
@smithfarm I guess you use the cephfs_* pools created by deepsea? Maybe the pool's pgs are not quite done creating?

**Comment by smithfarm** [comment]
> I guess you use the cephfs_* pools created by deepsea?

Yes.

> Maybe the pool's pgs are not quite done creating?

Will look into how I could determine this programatically, instead of blindly waiting 10 seconds (which seems to work well enough).

**Comment by swiftgist** [comment]
Is the cluster in a HEALTH_WARN state during the creation?  You might find

salt 'admin.ceph' wait.until status=HEALTH_OK

useful.

**Comment by smithfarm** [comment]
> salt 'admin.ceph' wait.until status=HEALTH_OK

Neat. Could it provide updates every 5 seconds so one could monitor if it is making progress?

**Comment by smithfarm** [comment]
Fixed by #328
